README
Beschreibung zu FF-Control

usage:

	create a build dir
	#mkdir ff-build
	#cd ff-build
	then clone this git repo
	#git clone git://github.com/freifunk/ff-control.git
	copy the default config or use your own config
	#cp ff-control/default.config ./config
	change the config if you want
	#nano config
	run the buildscript
	#ff-control/scripts/update-build.sh
	go sleeping ;-)
	rebuild only one package for all boards do
	#ff-control/scripts/update-build-pkg.sh packetname
	e.g.
	#ff-control/scripts/update-build-pkg.sh olsrd
	build firmware images with Image Builder
	#ff-control/scripts/update-wib.sh
	
customize:
	#cd ff-build
	#mkdir -p feeds/my-package
	Create a openwrt Makefile see http://wiki.openwrt.org/doc/devel/packages
	#nano feeds/my-package/Makefile
	Add the "my-package" y|m Build-In or Modul (opkg) to config
	#echo 'make_options=$make_options" CONFIG_PACKAGE_my-package=m"' >> config
	or
	#nano config
	the you can build your package or a new router image
	#ff-control/scripts/update-build-pkg.sh my-package
	or
	#ff-control/scripts/update-build.sh

dnsfoo:
	create a dnskey
	#dnssec-keygen -a HMAC-MD5 -b 512 -n USER myhostname.pberg.freifunk.net.
	Bind:
	#grep Key ./Kmyhostname.pberg.freifunk.net.*.private
	is the secret and myhostname.pberg.freifunk.net. is the key name.
	add key section in named.conf
key myhostname.pberg.freifunk.net. {
    algorithm hmac-md5;
    secret "From your keyfile ";
};
	add to update policy in zone config
zone "pberg.freifunk.net" in {
        file "master/pberg.freifunk.net";
        type master;
        notify yes;
        update-policy {
                grant myhostname.pberg.freifunk.net. wildcard *.myhostname.pberg.freifunk.net.;
        };
};

	Mac OS X:
	#grep Key ./Kmyhostname.pberg.freifunk.net.*.private
	is the password and myhostname.pberg.freifunk.net. the username.
	settings-->share-->[edit] hostname-->myhostname.pberg.freifunk.net.
	
	OLSR:
	#dnssec-keygen -a HMAC-MD5 -b 512 -n USER pberg.freifunk.net
	#grep Key ./Kmyhostname.pberg.freifunk.net.*.private
	is the secret and myhostname.pberg.freifunk.net. is the key name.
	add key section in named.conf
key pberg.freifunk.net. {
    algorithm hmac-md5;
    secret "From your keyfile ";
};
	add to update policy in zone config forward and reverse
zone "pberg.freifunk.net" in {
        file "master/pberg.freifunk.net";
        type master;
        notify yes;
        update-policy {
                grant pberg.freifunk.net. wildcard *.pberg.freifunk.net.;
        };
};
zone "8.3.0.5.0.7.4.0.1.0.0.2.ip6.arpa" IN {
    type master;
    file "master/2001:470:5038";
    update-policy {
        grant pberg.freifunk.net wildcard *.8.3.0.5.0.7.4.0.1.0.0.2.ip6.arpa.;
    };
};
	edit  scripts/dns-bonjour-geo.sh
	modify the first 20 lines
	#OLSR Router IPv6 Adresse
ROUTER="2001:470:6c:393::2"
#DNS Master Servername or IPv6 Adresse
SERVER="::1"
#DNS Zonenenname for AAAA entry's
ZONENAME="pberg.freifunk.net."
#DNS Reverse IPv6 Zone for PTR revnibbles.arpa entry's
ZONENET="2001:470:5038"
#TODO Automate
#uci show olsrd
#olsrd.@LoadPlugin[4].library=olsrd_nameservice.so.0.3
#olsrd.@LoadPlugin[4].hosts_file=/var/etc/hosts.olsr
HOSTS="/var/etc/hosts.olsr.ipv6"
TTL="240"
#Key file from dnssec-keygen output
KEYFILE="/etc/named.d/Kpberg.freifunk.net.+157+33698"

	create a ssh key for access to your olsr router
	#ssh-keygen
	#scp root@myrouter.olsr:/etc/dropbear/authorized_keys ./
	#cat ~/.ssh/id_rsa.pub >> authorized_keys
	#scp authorized_keys root@myrouter.olsr:/etc/dropbear/
	then try run scripts/dns-bonjour-geo.sh
	scripts/dns-bonjour-pberg.sh is a script thet make basic dns entrys
	for the pberg.freifunk.net domain.

	


CHANGELOG
# Changelog

## Version 0.0.1

### Packages

[packages] Add batctl to list (86de22f)
[packages] add collectd-mod-uptime and -mod-memory (32962c8)
[packages] added alfred to package list (337c54c)
[packages] added mtr to minimal list (35ade53)
[packages] add freifunk-berlin-firewall-defaults (e474626)
[packages] add freifunk-berlin-statistics-defaults (3731064)
[packages] add olsrd-mod-txtinfo (7e627e4)
[packages] add olsrd-mod-txtinfo and olsrd-mod-dyn-gw (f59e432)
[packages] move default values for olsr and network in own packages (bfe776d)
[packages] add px5g (4e8680a)
[packages] refactoring and removing unnecessary entries (ec94dc8)
[packages] refactor packages (80413ae)
[packages] remove 6to4 (61bcec0)
[packages] remove duplicate olsrd-mod-jsoninfo package (8ac0530)
[packages] remove wpad package (a845e14)
[packages] remove wpad (pulled in by mpc85xx) (0e6b5b6)
[packages] remove alfred from ar71xx.config (488f240)
[packages] remove auto-ipv6-{gw,node} from ar71xx.config (7a0d190)
[packages] remove olsrd-plugin-txtinfo (fix #103) (b01b4ca)
[packages] rename package list 'minimal' to 'vpn' (f42dc9c)
[packages] replace ar71xx.config with sane new config (130b8e1)
[packages] Add default packages for dhcp and freifunk (646e295)
[packages] Added additional packages to configs (227c13d)
[packages] Added freifunk-berlin-openvpn-files and freifunk-policyrouting (77fcddc)
[packages] Added package tmux to configs (0efc3ca)
[packages] Adding python to the list of missing pakages. (Missing for docker ubuntu images) (15cd128)

### Config

[configs] Added rudimentary packages list (024cafb)
[configs] #7 added luci-app-firewall package (c4db051)
[configs] do not use IGNOE_ERRORS=m (ced9337)
[configs] update openwrt revision (latest change) (e6abd81)
[configs] update openwrt revision to latest revision (4a6fcbb)
[configs] add community-profiles to ar71xx.config (2090804)
[configs] added alfred as package fixed dyn_gw (9f6e774)
[configs] added mtr as package to build (684647a)
[configs] add freifunk-berlin-firewall-default package (e949442)
[configs] add lib-guard, statistics-default and olsrd-mod-txtinfo (ee367b8)
[configs] add om-watchdog for OM2P target (1ec3938)
[configs] add px5g (00d8cc5)
[configs] add px5g to ar71xx.config (402ec38)
[configs] add x86 as target (42ca4df)
[configs] build with SSP_SUPPORT/libssp (d88f13c)
[configs] enable DFS support in ar71xx.config (1541177)
[configs] fix #20 VERSIONOPT bug (484d68a)
[configs] honor users regdb configuration (3b609e4)
[configs] only build image builder for ar71xx (83dc48b)
[configs] remove ppp support and some kernel debug flags (268812b)
[configs] set version information (622576d)
[configs] strip polarssl (aa81db0)
[configs] update repo link to buildbot (8e685fd)
[configs] update routing and luci (8b9a534)
[configs] update ar71xx (3229ebe)
[configs] fix https duplicated commonname/issuer id problem (e703655)
[configs] use barrier breaker 14.07 branch (92a31ed)
[configs] Add minimal package lists for 4MB flash devices (09bf78c)
[configs] add OM2P boards to ar71xx profiles (fbacfc5)
[configs] Add support for different default packages list (92f8ba9)
[configs] Add target 'x86' (aea021f)
[configs] update ar71xx.config to current openwrt revision (92a310e)
[configs] update luci feed and remove obsolete map patch (d0d4284)
[configs] Update OpenWRT version (e5737d7)

### Feeds

[feeds] adapt feeds for freifunk (e4ef5a6)
[feeds] add support for statistics (e1d5f4b)
[feeds] bugfixes for ffwizard and freifunk-defaults * freifunk-defaults - add missing firewall rules * ffwizard-berlin - set wan type to bridge for private APs (63d7cdf)
[feeds] change page order of optional stuff and wireless (a31ed55)
[feeds] configure olsrd with defaults values from community_profile (ce2ffed)
[feeds] update luci feed and rebase/remove patches (087fc30)
[feeds] update revision of packages-berlin remove defaults from wizard (e714ddd1)
[feeds] ffwizard - activate dhcpv6+ra server mode for odhcpd for dhcp network (787b311)
[feeds] ffwizard-berlin: firewall - add dhcp to freifunk zone (95f4a51)
[feeds] ffwizard-berlin - only configure statistics if installed (8963389)
[feeds] ffwizard-berlin support wifi iface5, osm tiles (9f6c437)
[feeds] ffwizard-berlin - use HT40 only for channels 36..100 (2e353a9)
[feeds] fix copy/paste error in uci-defaults of ffwizard-berlin (9bf4324)
[feeds] fix zoom level for map in ffwizard-berlin (0696a93)
[feeds] olsrd defaults - add RtTablePriority to fix netlink errors (9f057b3)
[feeds] pin down all feeds to a specific commit (886ccf9)
[feeds] update berlin-packages always enable policy routing (65e57cb)
[feeds] update berlin-packages ix openwifimap, wireless ap config (57a8c1b)
[feeds] update berlin-packages update for smartgw and dyngw (470aea0)
[feeds] updated packages_berlin add lib guard fix ipinfo (af6aa36)
[feeds] update ffwizard-berlin fix countryCode,olsr, qos (821a29f)
[feeds] update ffwizard-berlin wizard with map, private ap, vap (7e2652b)
[feeds] update luci feed (3b4e592)
[feeds] update package_berlin (9919dd4)
[feeds] update package_berlin change default ip to 192.168.42.1, add static
[feeds] update package_berlin remove p2pblock, added switch-ports to br-dhcp (ff83bd4)
[feeds] update packages-berlin no ipip tunnel if uplink (e336804)
[feeds] update packages_berlin update openvpn setup (b79c5d4)
[feeds] update packages_berlin toggle stats in wizard (c9d419f)
[feeds] use dyngw instead of dyngw_plain+ff_olsr_gwcheck (ec593c6)
[feeds] use ffwizard-berlin (1580033)

### Patches

[patches] #18 redirect to ffwizard without login (5732ad5)
[patches] #18 redirect to freifunk wizard on first boot (3085cc6)
[patches] #41 change ssid to berlin.freifunk.net (5d4f9e1)
[patches] add barrier breaker patches (38aa6b7)
[patches] add bind ipv6 only patch for olsrd txtinfo and jsoninfo (6cb7a33)
[patches] add default values for olsrd to community_profile (90e8e97)
[patches] added freifunk gwcheck jshn patch (ac0327a)

scripts/update-dns.sh
#!/bin/sh

export MAILTO=""
#DNS Master Servername or IPv6 Adresse
SERVER="81.169.174.233"
#DNS Zonenenname for AAAA entry's
ZONENAME="pberg.freifunk.net."
#DNS Reverse IPv6 Zone for PTR revnibbles.arpa entry's
ZONENET="2001:470:5038"
TTL="240"
#Key file from dnssec-keygen output
KEYFILE="Kpberg.freifunk.net.+157+33698"
TIME=$(date --rfc-3339=ns)

update() {
      SERVER="$1"
      ENTRY="$2"
      RECORD="$3"
      NAME="$4"
      DATE="$(date)"
      echo "update: $ENTRY $RECORD $NAME"
      /usr/bin/nsupdate -v -k "$KEYFILE" <<EOF
server $SERVER
update delete $ENTRY $RECORD
update add $ENTRY $TTL $RECORD $NAME
send
EOF
}

updatesrv() {
      SERVER="$1"
      ENTRY="$2"
      RECORD="$3"
      PORT="$4"
      NAME="$5"
      TXT="$6"
      echo "updatesrv:  $ENTRY $RECORD $PORT $NAME $TXT"
      /usr/bin/nsupdate -v -k "$KEYFILE" <<EOF
server $SERVER
update delete $ENTRY $RECORD
update delete $ENTRY TXT
update add $ENTRY $TTL TXT "$TXT $DATE"
update add $ENTRY $TTL $RECORD 0 0 $PORT $NAME
send
EOF
}

updateptr() {
      SERVER="$1"
      ENTRY="$2"
      RECORD="$3"
      NAME="$4"
      echo "updateptr: $ENTRY $RECORD $NAME"
      /usr/bin/nsupdate -v -k "$KEYFILE" <<EOF
server $SERVER
update add $ENTRY $TTL $RECORD $NAME
send
EOF
}

updateservices(){
	name="$1"
	services="$2"
	for i in $services ; do
		OIFS=$IFS;IFS=:
		set $i;srv=$1;port=$2
		IFS=$OIFS
		updateptr "$SERVER" "$srv.$ZONENAME" "PTR" "$name.$srv.$ZONENAME"
		updatesrv "$SERVER" "$name.$srv.$ZONENAME" "SRV" "$port" "$name.$ZONENAME" "$name"
	done
}

#update "$SERVER" "testing.$ZONENAME" "A" "81.169.174.233"
#update "$SERVER" "testing.$ZONENAME" "AAAA" "2a01:238:4324:9700:7412:6a5e:7261:31de"
#updateservices "testing" "_ssh._tcp:22 _sftp-ssh._tcp:22 _http._tcp:80"

update "$SERVER" "druschba.$ZONENAME" "A" "77.64.239.35"
